.github/actions/update-axe-core-v1/src/run.ts

```diff
@@ -0,0 +1,104 @@
+import type { Core } from './types'
+import { glob, type GlobOptions } from 'glob'
+import { getExecOutput } from '@actions/exec'
+import path from 'path'
+import type { getPackageManagerReturn } from './types'
+
+export default async function run(
+  core: Core,
+  getPackageManager: (dirPath: string) => getPackageManagerReturn,
+  cwd?: string
+) {
+  try {
+    const { stdout: latestAxeCoveVersion } = await getExecOutput('npm', ['info', 'axe-core', 'version'])
+    core.info(`latest axe-core version ${latestAxeCoveVersion}`)
+
+    // npm and yarn workspaces will have a lock file at the root
+    // of the project
+    const rootPackageManager = await getPackageManager('./')
+    core.info(`root package manager detected as ${rootPackageManager}`)
+
+    const packages = await glob('**/package.json', {
+      absolute: true,
+      cwd,
+      ignore: '**/node_modules/**/package.json'
+    })
+    let installedAxeCoreVersion: string | null = null
+
+    for (const filePath of packages) {
+      core.info(`package.json found in ${filePath}`)
+      const pkg = await import(filePath as string)
```

Comment:
I wish it weren't so, but the return type of Glob seems to be `Array<string | Path>` which is just 🤦

Reply:
That should only be the case with if you pass `withFileTypes` does it resolve to `Promise<string[] | path[]>`, otherwise the type should resolve with `string[]`

Reply:
Weird. It was totally throwing errors before but now it's not? 🤷

Reply:
```ts
if (language.isTypeScript && user.steve) {
	console.error('you shall not pass!')
}
```

Reply:
Every time